orchagent/routeorch.cpp

```diff
@@ -1027,8 +1163,97 @@ bool RouteOrch::addRoute(sai_object_id_t vrf_id, const IpPrefix &ipPrefix, const
         route_attr.value.oid = next_hop_id;
 
         /* Default SAI_ROUTE_ATTR_PACKET_ACTION is SAI_PACKET_ACTION_FORWARD */
-        sai_status_t status = sai_route_api->create_route_entry(&route_entry, 1, &route_attr);
-        if (status != SAI_STATUS_SUCCESS)
+        object_statuses.emplace_back();
+        sai_status_t status = gRouteBulker.create_entry(&object_statuses.back(), &route_entry, 1, &route_attr);
+        if (status == SAI_STATUS_ITEM_ALREADY_EXISTS)
+        {
+            SWSS_LOG_ERROR("Failed to create route %s with next hop(s) %s: already exists in bulker",
```

Comment:
Is this really an ERROR case? I don't have an issue with the logic, but if this is something that could come up commonly in practice then I don't know if we should be printing errors to the syslog.

Reply:
It is a logic error. `m_toSync` will not have duplicated SET routes because of its design.

---
In reply to: [417493477](https://github.com/Azure/sonic-swss/pull/1238#discussion_r417493477) [](ancestors = 417493477)